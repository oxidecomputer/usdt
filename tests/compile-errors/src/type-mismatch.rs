#![feature(asm)]
#![cfg_attr(target_os = "macos", feature(asm_sym))]

usdt::dtrace_provider!("../../../tests/compile-errors/providers/type-mismatch.d");

fn main() {
    let bad: f32 = 0.0;
    mismatch::bad!(|| (bad));
}
