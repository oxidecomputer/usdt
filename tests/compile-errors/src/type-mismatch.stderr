error[E0277]: the trait bound `f32: Borrow<u8>` is not satisfied
  --> $DIR/type-mismatch.rs:15:1
   |
15 | usdt::dtrace_provider!("../../../tests/compile-errors/providers/type-mismatch.d");
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Borrow<u8>` is not implemented for `f32`
...
19 |     mismatch::bad!(|| (bad));
   |     ------------------------ in this macro invocation
   |
note: required by a bound in `__usdt_private_mismatch_bad_type_check`
  --> $DIR/type-mismatch.rs:15:1
   |
15 | usdt::dtrace_provider!("../../../tests/compile-errors/providers/type-mismatch.d");
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `__usdt_private_mismatch_bad_type_check`
...
19 |     mismatch::bad!(|| (bad));
   |     ------------------------ in this macro invocation
   = note: this error originates in the macro `mismatch::bad` (in Nightly builds, run with -Z macro-backtrace for more info)
